inbound_test.go

```diff
@@ -142,3 +142,44 @@ func TestInboundConnection_CallOptions(t *testing.T) {
 		require.NoError(t, err, "Call through proxy failed")
 	})
 }
+
+func TestBlackhole(t *testing.T) {
+	ctx, cancel := NewContext(testutils.Timeout(time.Hour))
+
+	testutils.WithTestServer(t, nil, func(server *testutils.TestServer) {
+		serviceName := server.ServiceName()
+		handlerName := "test-handler"
+
+		server.Register(HandlerFunc(func(ctx context.Context, inbound *InboundCall) {
+			// cancel client context in handler so the client can return after being blackholed
+			defer cancel()
+
+			c, _ := InboundConnection(inbound)
+			require.NotNil(t, c)
+
+			state := c.IntrospectState(&IntrospectionOptions{})
+			require.Equal(t, 1, state.InboundExchange.Count, "expected exactly one inbound exchange")
+
+			// blackhole request
+			inbound.Response().Blackhole()
+
+			// give time for exchange to cleanup
+			require.True(t, testutils.WaitFor(10*time.Millisecond, func() bool {
+				state = c.IntrospectState(&IntrospectionOptions{})
+				return state.InboundExchange.Count == 0
+			}),
+				"expected no inbound exchanges",
+			)
+
+		}), handlerName)
+
+		clientCh := server.NewClient(nil)
+		defer clientCh.Close()
+
+		_, _, _, err := raw.Call(ctx, clientCh, server.HostPort(), serviceName, handlerName, nil, nil)
+		require.Error(t, err, "expected call error")
+
+		errCode := GetSystemErrorCode(err)
+		assert.Equal(t, ErrCodeCancelled, errCode, "expected cancelled error code, got: %q", errCode)
```

Comment:
no need for "got %v", `assert` will print the value it expected and what it got. can just make error message something like "got unexpected error code"

Reply:
As discussed in person, this would output the `byte` value (eg. `0x02`) since `SystemErrCode` is an alias of `byte`. I'll update with a comment to indicate why I wrote it out this way.